File: javapatterns/src/main/java/org/iut/nantes/factories/cours/Cours.java
package org.iut.nantes.factories.cours;

abstract class Cours
{
  protected String matiere;

  public String getMatiere() { return matiere; }
}

class CoursInfo extends Cours
{
  public CoursInfo()
  {
    matiere = "info";
  }
}

class CoursMaths extends Cours
{
  public CoursMaths()
  {
    matiere = "maths";
  }
}

class CoursChimie extends Cours
{
  public CoursChimie()
  {
    matiere = "chimie";
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/factories/cours/Cours.kt
package org.iut.nantes.factories.cours

abstract class Cours {
    var matiere: String? = null
        protected set
}

class CoursInfo : Cours() {
    init {
        matiere = "info"
    }
}

class CoursMaths : Cours() {
    init {
        matiere = "maths"
    }
}

class CoursChimie : Cours() {
    init {
        matiere = "chimie"
    }
}